app/lib/shared/email.dart

```diff
@@ -0,0 +1,145 @@
+    if (match != null) {
+      name = match.group(1).trim();
+      email = match.group(2).trim();
+    } else if (value.contains('@')) {
+      final List<String> parts = value.split(' ');
+      for (int i = 0; i < parts.length; i++) {
+        if (parts[i].contains('@') &&
+            parts[i].contains('.') &&
+            parts[i].length > 4) {
+          email = parts[i];
+          parts.removeAt(i);
+          name = parts.join(' ');
+          break;
+        }
+      }
+    }
+    if (name != null && name.isEmpty) {
+      name = null;
+    }
+    if (email != null && email.isEmpty) {
+      email = null;
+    }
+    return new EmailAddress(name, email);
+  }
+
+  bool get isEmpty => name == null && email == null;
+
+  @override
+  String toString() {
+    if (isEmpty) return null;
+    if (email == null) return name;
+    if (name == null) return email;
+    return '$name <$email>';
+  }
+}
+
+/// Represents an email message the site will send.
+class EmailMessage {
+  final EmailAddress from;
+  final List<EmailAddress> recipients;
+  final String subject;
+  final String bodyText;
+
+  EmailMessage(this.from, this.recipients, this.subject, String bodyText)
+      : bodyText = reflowBodyText(bodyText);
+}
+
+/// Parses the body text and splits the [input] paragraphs to [lineLength]
+/// character long lines.
+String reflowBodyText(String input, {int lineLength = 72}) {
+  Iterable<String> reflowLine(String line) sync* {
+    if (line.isEmpty) {
+      yield line;
+      return;
+    }
+    while (line.length > lineLength) {
+      int firstSpace = line.lastIndexOf(' ', lineLength);
+      if (firstSpace < 20) {
+        firstSpace = line.indexOf(' ', lineLength);
+      }
+      if (firstSpace != -1) {
+        yield line.substring(0, firstSpace);
+        line = line.substring(firstSpace).trim();
+      } else {
+        yield line;
+        return;
+      }
+    }
+    if (line.isNotEmpty) {
+      yield line;
+    }
+  }
+
+  return input.split('\n').expand(reflowLine).join('\n');
+}
+
+/// Creates the [EmailMessage]s that we be sent on new package upload.
+EmailMessage createPackageUploadedEmail({
+  @required String packageName,
+  @required String packageVersion,
+  @required String uploaderEmail,
+  @required List<EmailAddress> authorizedUploaders,
+  @required List<EmailAddress> authors,
+}) {
+  final subject = 'Package upload on pub: $packageName $packageVersion';
+  final bodyText = new StringBuffer()
+    ..writeln('Dear package maintainer,')
+    ..writeln()
+    ..writeln(
+        '$uploaderEmail uploaded a new version of package $packageName: $packageVersion')
+    ..write('\n');
+
+  if (authors.isNotEmpty) {
+    bodyText.writeln('The listed authors are the following:');
+    authors.forEach((author) {
+      bodyText.writeln('- $author');
+    });
+    bodyText.writeln();
+  }
+
+  bodyText
+    ..writeln(
+        'If you think this is a mistake or fraud, contact us at ${_defaultFrom.email}')
```

Comment:
don't write that email, file an issue on github:
https://github.com/dart-lang/pub-dartlang-dart/issues